frontend/packages/metal3-plugin/src/components/dashboard/inventory-card.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import { DashboardItemProps } from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { DashboardCard } from '@console/internal/components/dashboard/dashboard-card';
+import { DashboardCardBody } from '@console/internal/components/dashboard/dashboard-card/card-body';
+import { DashboardCardHeader } from '@console/internal/components/dashboard/dashboard-card/card-header';
+import { DashboardCardTitle } from '@console/internal/components/dashboard/dashboard-card/card-title';
+import {
+  InventoryItem,
+  Status,
+} from '@console/internal/components/dashboard/inventory-card/inventory-item';
+import { InventoryStatusGroup } from '@console/internal/components/dashboard/inventory-card/status-group';
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { MachineModel } from '@console/internal/models';
+import { getNamespace } from '@console/shared';
+
+import { getHostStorage, getHostNICs, getHostCPU, getHostMachineName } from '../../selectors';
+import { getInventoryQueries, HostQuery } from './queries';
+
+const getResources = (namespace: string, machineName: string): FirehoseResource[] => [
+  {
+    isList: false,
+    namespace,
+    name: machineName,
+    kind: referenceForModel(MachineModel),
+    prop: 'machine',
+  },
+];
+
+export const InventoryCard: React.FC<InventoryCardProps> = ({
+  obj,
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+}) => {
+  const namespace = getNamespace(obj);
+  const machineName = getHostMachineName(obj);
+  const k8sResources = getResources(namespace, machineName);
+  const machineAddresses = _.get(resources.machine, 'data.status.addresses', []);
+  const machineIP = _.get(machineAddresses.find((addr) => addr.type === 'InternalIP'), 'address');
+
+  React.useEffect(() => {
+    k8sResources.forEach((r) => watchK8sResource(r));
+    return () => {
+      k8sResources.forEach((r) => stopWatchK8sResource(r));
+    };
+  }, [watchK8sResource, stopWatchK8sResource, k8sResources]);
```

Comment:
`k8sResources` will be a different object on every update of `obj`. It would be better if you add `namespace` and `machineName` as dependency instead of `k8sResources`.
```
React.useEffect(() => {
    const k8sResources = getResources(namespace, machineName);
    k8sResources.forEach((r) => watchK8sResource(r));
    return () => {
      k8sResources.forEach((r) => stopWatchK8sResource(r));
    };
  }, [watchK8sResource, stopWatchK8sResource, namespace, machineName]);
```